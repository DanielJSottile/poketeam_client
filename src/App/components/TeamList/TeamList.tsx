import React, { useContext, FunctionComponent } from 'react';
import Team from '../Team';
import LazyLoader from '../Loaders/LazyLoader';
import TeamListPagination from './TeamListPagination';
import GeneralContext from '../../contexts/GeneralContext';
import styles from './TeamList.module.scss';
import { PokemonTeam } from '../../@types';

type TeamListProps = {
  /** determines whether the list is public or private */
  isPublic: boolean;
};

const TeamList: FunctionComponent<TeamListProps> = ({
  isPublic,
}): JSX.Element => {
  const { userTeams, currentClickedFolder, publicTeams } =
    useContext(GeneralContext);

  const publicTeamList = publicTeams.map((team: PokemonTeam, i) => {
    return <Team isPublic key={i} id={`${team.team_name}`} team={team} />;
  });

  const privateTeamList = userTeams
    .filter(
      (team: PokemonTeam) => team.folder_id === Number(currentClickedFolder.id)
    )
    .map((team: PokemonTeam, i) => {
      return (
        <Team isPublic={false} id={`${team.team_name}`} key={i} team={team} />
      );
    });

  const teamList = isPublic ? publicTeamList : privateTeamList;

  const renderLazyOrMessage = () =>
    isPublic ? (
      <LazyLoader
        containerClass={styles['pokeball-div']}
        messageClass={styles['hint']}
        message={'(Hint: There May Be No Teams)'}
      />
    ) : (
      <h3>
        Either No Folder is Selected Or There Are No Teams! Make Teams with
        Pokemon!
      </h3>
    );

  return (
    <>
      {isPublic && <TeamListPagination />}
      {teamList.length ? teamList : renderLazyOrMessage()}
    </>
  );
};

export default TeamList;
