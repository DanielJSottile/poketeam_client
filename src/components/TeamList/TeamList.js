import React, { Component, Fragment } from 'react'

export default class TeamList extends Component {
  render() {
    return (
      <Fragment>
        
      </Fragment>
    );
  };
};


