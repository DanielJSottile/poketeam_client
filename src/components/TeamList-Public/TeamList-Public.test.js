import React from 'react';
import ReactDOM from 'react-dom';
//import { render } from '@testing-library/react';
import TeamList from './TeamList-Public';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<TeamListPublic/>', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<TeamList />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});