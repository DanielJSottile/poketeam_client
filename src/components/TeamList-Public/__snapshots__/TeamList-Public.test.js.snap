// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamListPublic/> renders without crashing 1`] = `
<TeamList>
  <h3>
    There Are No Teams!  Make Teams with Pokemon!
  </h3>
</TeamList>
`;
